src/Http/Routing/src/Matching/DfaMatcher.cs

```diff
@@ -68,6 +69,26 @@ public sealed override Task MatchAsync(HttpContext httpContext, EndpointSelector
                 Logger.CandidatesFound(_logger, path, candidates);
             }
 
+            var policyCount = policies.Length;
+
+            // This is a fast path for single candidate, 0 policies and default selector
+            if (policyCount == 0 && candidateCount == 1 && _selector is DefaultEndpointSelector)
```

Comment:
How fast is `is`? Is it worth doing this check once in the constructor and setting a bool on DfaMatcher to check here?

Reply:
Good idea.

Reply:
yeah it's probably more consistent to make this a flag on candidates if possible.

Reply:
eh, might not actually be a good fit for flags.